packages/salesforcedx-vscode-core/package.json

```diff
@@ -83,35 +83,43 @@
       "explorer/context": [
         {
           "command": "sfdx.force.apex.class.create",
-          "when": "explorerResourceIsFolder && sfdx:project_opened"
+          "when":
+            "explorerResourceIsFolder && resourceFilename != aura && resourceFilename != lightningcomponents && sfdx:project_opened"
```

Comment:
It would be nice if we could create a re-usable compound when-clause. I don't think they have that support yet though.